### ISSUE TITLE
Problem with Complex Regex Expression Handling in Metabase with Athena

### ISSUE THREAD

**Issue by omkar-work** [metabase/metabase#33878]
### Describe the bug

**Issue Description:**
Hello everyone,

I've encountered an issue while working on a native query in Athena that involves using a regex expression with Metabase. Here's the query I'm working with:

```sql
select * from <table> where regexp_like(col_name, {{match_filter}})
```

The `match_filter` is a string filter that determines the results based on the regex match. When I use a simple string like 'trial_text' as the filter, everything works perfectly. However, the problem arises when I attempt to use a more complex regex expression like '^(?!.\btrial_text\b).$', which is intended to exclude anything containing 'trial_text.' At this point, I encounter an error in Metabase, and the error message is as follows:

```
[Simba][AthenaJDBC](100071) An error has been thrown from the AWS Athena client. line 49:46: mismatched input '^'. Expecting: ',', <expression> [Execution ID not available]
```

Interestingly, the same query runs without any issues directly in Athena. It appears that there may be an issue with how Metabase handles complex expressions with special characters.

Any assistance in resolving this issue would be greatly appreciated.

### To Reproduce

To clarify the situation further, I have two example queries:

1. ```sql
   select * from t1 where regexp_like(col, 'trial_text')
   ```

2. ```sql
   select * from t1 where regexp_like(col, '^(?!.*\btrial_text\b).*$')
   ```

I am attempting to parameterize the second parameter using a text filter called `match_filter` like this:

```sql
select * from t1 where regexp_like(col, {{match_filter}})
```

The first query works perfectly fine when I use 'trial_text' as input. However, it fails to work when I use '^(?!.\btrial_text\b).$' as input.

### Information about your Metabase installation

```JSON
Latest version of metabase on chrome
```

### Severity

Medium

**Comment by omkar-work** [comment]
@flamber - Can u please help here

**Comment by luizarakaki** [comment]
We need to identify if this is a problem in Metabase or the JDBC driver we are using.

**Comment by luizarakaki** [comment]
@omkar-work if you don't use the variable here and run only 
```
select * from t1 where regexp_like(col, '^(?!.*\btrial_text\b).*$')
```
in Metabase, does it fail?

I'm trying to exclude parameter substitution from the issue.

**Comment by omkar-work** [comment]
Yes, this works fine in metabase without parameter substitution.


 
  
  
   Thanks,
  
  
   Omkar
  
  
   
   
   
  
 
 
  
  
 

 
  
 
   On Tue, Oct 03, 2023, 11:43 pm Luiz Arakaki ***@***.***> wrote: 
   
  
  
   
  @omkar-work if you don't use the variable here and run only 
  select * from t1 where regexp_like(col, '^(?!.*\btrial_text\b).*$')
 
  in Metabase, does it fail? 
  I'm trying to exclude parameter substitution from the issue. 
  —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***> 
  [
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/metabase/metabase/issues/33878#issuecomment-1745483185",
"url": "https://github.com/metabase/metabase/issues/33878#issuecomment-1745483185",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by luizarakaki** [comment]
Got it... So there is something going wrong when substituting the param there... Likely on our side then
